lib/tbot/config/service_workload_identity_aws_ra.go

```diff
@@ -0,0 +1,170 @@
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package config
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws/arn"
+	"github.com/gravitational/trace"
+	"gopkg.in/yaml.v3"
+
+	"github.com/gravitational/teleport/lib/tbot/bot"
+)
+
+const (
+	WorkloadIdentityAWSRAType        = "workload-identity-aws-ra"
+	defaultAWSSessionDuration        = 6 * time.Hour
+	maxAWSSessionDuration            = 12 * time.Hour
+	defaultAWSSessionRenewalInterval = 1 * time.Hour
+)
+
+var (
+	_ ServiceConfig = &WorkloadIdentityAWSRAService{}
+	_ Initable      = &WorkloadIdentityAWSRAService{}
+)
+
+// WorkloadIdentityAWSRAService is the configuration for the
+// WorkloadIdentityAWSRAService
+type WorkloadIdentityAWSRAService struct {
+	// Selector is the selector for the WorkloadIdentity resource that will be
+	// used to issue WICs.
+	Selector WorkloadIdentitySelector `yaml:"selector"`
+	// Destination is where the credentials should be written to.
+	Destination bot.Destination `yaml:"destination"`
+
+	// RoleARN is the ARN of the role to assume.
+	// Example: `arn:aws:iam::123456789012:role/example-role`
+	// Required.
+	RoleARN string `yaml:"role_arn"`
+	// ProfileARN is the ARN of the Roles Anywhere profile to use.
+	// Example: `arn:aws:rolesanywhere:us-east-1:123456789012:profile/0000000-0000-0000-0000-00000000000`
+	// Required.
+	ProfileARN string `yaml:"profile_arn"`
+	// TrustAnchorARN is the ARN of the Roles Anywhere trust anchor to use.
+	// Example: `arn:aws:rolesanywhere:us-east-1:123456789012:trust-anchor/0000000-0000-0000-0000-000000000000`
+	// Required.
+	TrustAnchorARN string `yaml:"trust_anchor_arn"`
+	// Region is the AWS region to use.
+	// Example: `us-east-1`
+	// Must be set here or in the environment or AWS config using the
+	// `AWS_REGION` environment variable. If set here, this will override the
+	// environment or AWS config.
+	Region string `yaml:"region"`
+
+	// SessionDuration is the duration of the resulting AWS session and
+	// credentials. This may be up to 12 hours. When unset, this defaults to
+	// 6 hours.
+	SessionDuration time.Duration `yaml:"session_duration"`
+	// SessionRenewalInterval is the interval at which the session should be
+	// renewed. This should be less than the session duration. When unset, this
+	// defaults to 1 hour.
+	SessionRenewalInterval time.Duration `yaml:"session_renewal_interval"`
```

Comment:
As these are equivalent to `credential_ttl` and `renewal_interval` we could use those field name instead for symmetry with the other services. I'm guessing we've opted not to do that though, because these names more clearly align with the AWS nomenclature?

Reply:
Makes sense, thanks for the explanation!